shared/Utils.js

```diff
@@ -3,5 +3,50 @@
 export default {
 	stripHtmlTags (inputString) {
 		return inputString.replace(/<\/?[^>]+(>|$)/g, "");
+	},
+
+	/**
+	 * Generate the navigation menu based on the response from the Wordpress API.
+	 * The retun menu array will:
+	 * 1. Exclude all pages whose menu_order === 0;
+	 * 2. Append "News" and "Views" menu items to the end.
+	 * @param {Array<Object>} sitePages - All menu pages sorted by menu_order.
+	 * @param {Boolean} includeHomeItem - A flag indicating if the home menu item should be included.
+	 * @return {Array<Object>} An array of object including the slug, menu title and href.
+	 */
+	generateNavMenu (sitePages, includeHomeItem) {
+		// Don't include the menu item for "home" by default
+		includeHomeItem = includeHomeItem === undefined ? false : includeHomeItem;
+		const menuItems = [];
+		const homeMenuItem = [];
+		const itemsToAppend = [{
```

Comment:
I think this needs a more descriptive name. I'm not sure why these menu items are hard coded and not included with `sitePages`.

Reply:
"News" and "Views" doesn't have a page defined in the WP back-end. Their page contents are populated at the WeCount website based on posts published in these categories.

Reply:
Maybe refer to these as dynamic pages or something? Is there a way to source the reference from outside of this function and pass it in? That way if we add/remove/modify them we won't have to touch this function.

Reply:
This function is called to generate nav menus on the header and the footer. This logic of appending "News" and "Views" is applicable to both places. I think the goal of this function is to be able to deal with this use case at one common place. Passing in these items as an argument would defeat this purpose.

Reply:
I guess what I meant is that if we convert Views for example to be a static page, we'll also have to update this function to remove it. Or if we change the Title. This function will have to be updated. I guess that last point goes for Home as well.

If these don't seem like issues that will come up, or are not things we need to deal with now, then we can punt the discussion for later.

Reply:
When we convert News and Views to static pages, these menu items still need to be appended. Creating static page content doesn't change the fact that these pages don't have landing pages defined in WP. 

In terms of the title, yes, if the design changes the title, this function needs to be adjusted. Since there isn't an API fetch telling us what these titles should be, hardcoding them here is what I can come up with. At least only this function needs to be adjusted when there's a title change.

About the title for "Home", what I received from API for this title is `Creating an inclusive data ecosystem`, which doesn't give any hint that it's a home page. And it also doesn't match what's on the design. That's why the title is hardcoded to "Home" based on its slug.

Reply:
Let's punt this for now. Might be useful to migrate some of this discussion into comments in the code. I'm thinking the issue with `Home` would be a good idea to leave a comment about.